File: sdks/python/bolt_prediction_sdk/models/treatment_plan_schema.py
"""
TreatmentPlanSchema 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class TreatmentPlanSchema(SDKBaseModel):
    """处置方案"""

    plan_name: Optional[Any] = Field(description="方案名称", default=None)
    steps: Optional[List[TreatmentStepSchema]] = Field(description="处置步骤列表", default=None)
    materials: Optional[Any] = Field(description="所需材料", default=None)
    estimated_cost: Optional[Any] = Field(description="预估成本", default=None)
    difficulty_level: Optional[Any] = Field(description="难度等级 easy/medium/hard", default=None)
    personnel_required: Optional[Any] = Field(description="所需人员", default=None)

File: sdks/java/src/main/java/com/boltprediction/sdk/model/TreatmentPlanSchema.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 处置方案 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class TreatmentPlanSchema {

    @JsonProperty("plan_name")
    private Object planName;

    @JsonProperty("steps")
    private List<TreatmentStepSchema> steps;

    @JsonProperty("materials")
    private Object materials;

    @JsonProperty("estimated_cost")
    private Object estimatedCost;

    @JsonProperty("difficulty_level")
    private Object difficultyLevel;

    @JsonProperty("personnel_required")
    private Object personnelRequired;

    public Object getPlanName() {
        return planName;
    }

    public void setPlanName(Object planName) {
        this.planName = planName;
    }

    public List<TreatmentStepSchema> getSteps() {
        return steps;
    }

    public void setSteps(List<TreatmentStepSchema> steps) {
        this.steps = steps;
    }

    public Object getMaterials() {
        return materials;
    }

    public void setMaterials(Object materials) {
        this.materials = materials;
    }

    public Object getEstimatedCost() {
        return estimatedCost;
    }

    public void setEstimatedCost(Object estimatedCost) {
        this.estimatedCost = estimatedCost;
    }

    public Object getDifficultyLevel() {
        return difficultyLevel;
    }

    public void setDifficultyLevel(Object difficultyLevel) {
        this.difficultyLevel = difficultyLevel;
    }

    public Object getPersonnelRequired() {
        return personnelRequired;
    }

    public void setPersonnelRequired(Object personnelRequired) {
        this.personnelRequired = personnelRequired;
    }

}